### DISCUSSION TITLE
Assembly location in Revit 25

### DISCUSSION THREAD

**Discussion by dinuzeros** [chuongmep/RevitAddInManager#56] [General]
Is there any way to access the location of an addin loaded into the addin manager in Revit 25? In previous revit versions, I accessed the location property of the assembly, but (probably thanks to the way net8 handles assemblies) in Revit 25 the location is null.

**Comment by chuongmep** [comment]
Hi @dinuzeros , can show me which code you are trying ?

**Comment by dinuzeros** [comment]
```
var myAddin = Assembly.GetExecutingAssembly();
TaskDialog.Show("Location", myAddin.Location);

var myAddin2 = MethodBase.GetCurrentMethod().DeclaringType.Assembly;
TaskDialog.Show("Location", myAddin2.Location);
```
Both of those give me the path to the temp folder, where Addinmanager Copies the Asssembly to run, in Revit 2023, but an Empty string (Seems like i was wrong about the null) when running the command in Revit 2025.

(Besides that simple example: we have a bunch of csv that we have in the same directory as our addin and to load them, we need to know, where the addin is. :) )

**Reply by chuongmep** [reply]
I see, the reason is upgrade version 25 of add-in manager don't copy to temp folder anymore, we use [AssemblyLoadContext](https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext) to load assemblies and dependencies, and then unload it when it excuted. That why you don't see any location path of `ExecutingAssembly` anymore.

**Reply by dinuzeros** [reply]
Okay makes sense. Thanks a lot for the context.

**Reply by chuongmep** [reply]
No problem, enjoy your coding.

**Reply by gklein30** [reply]
> I see, the reason is upgrade version 25 of add-in manager don't copy to temp folder anymore, we use [AssemblyLoadContext](https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext) to load assemblies and dependencies, and then unload it when it excuted. That why you don't see any location path of `ExecutingAssembly` anymore.

This seem inaccurate. It appears that the location is not available because the AddinManagerBase.RunActiveCommand method for 2025 is loading the assembly from stream. This has less to do with the AssemblyLoadContext and more do with how the assembly is loaded. What is the rational behind loading the assemblies from stream rather than loading it from path?
